packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/test/Providers/ClientProviders/ClientProviderTests.cs

```diff
@@ -334,6 +334,115 @@ public void TestSubClientAccessorFactoryMethods(InputClient client, bool hasSubC
+            => new InputClient(
+                TestClientName,
+                "TestClient description",
+                [
+                    new InputOperation(
+                        "Operation",
+                        null,
+                        "Operation",
+                        null,
+                        "public",
+                        [
+                            new InputParameter(
+                                "queryParam",
+                                "queryParam",
+                                "queryParam",
+                                new InputEnumType(
+                                    "InputEnum",
+                                    "InputEnum",
+                                    "public",
+                                    null,
+                                    "InputEnum",
+                                    InputModelTypeUsage.Input,
+                                    new InputPrimitiveType(InputPrimitiveTypeKind.String, "string", "string", null, null),
+                                    [
+                                        new InputEnumTypeValue("value1", "value1", "value1"),
+                                        new InputEnumTypeValue("value2", "value2", "value2"),
+                                    ],
+                                    true),
+                                RequestLocation.Query,
+                                defaultValue: null,
+                                InputOperationParameterKind.Method,
+                                isRequired: false, false, false, false, false, false, false, null, null)
+                        ],
+                        [
+                            new OperationResponse(
+                                [200],
+                                null,
+                                BodyMediaType.None,
+                                [],
+                                false,
+                                [])
+                        ],
+                        "GET",
+                        BodyMediaType.None,
+                        "/foo",
+                        "/foo",
+                        null,
+                        null,
+                        true,
+                        null,
+                        null,
+                    true,
+                    true,
+                        "Operation")
+                ],
+                [],
+                null);
+
+        private class ValidateQueryParamDiffClientProvider : ClientProvider
+        {
+            public ValidateQueryParamDiffClientProvider(InputClient client)
+                : base(client)
+            {
+            }
+
+            protected override MethodProvider[] BuildMethods()
+            {
+                var method = base.BuildMethods().Where(m => m.Signature.Parameters.Any(p => p.Name == "queryParam" && p.Type.Name == "InputEnum" && !m.Signature.Name.EndsWith("Async"))).First();
```

Comment:
Could we have both the convenience and protocol methods?

Reply:
we want to validate the call from conv to protocol so having both does not make sense in this context.